### DISCUSSION TITLE
Update without Docker

### DISCUSSION THREAD

**Discussion by dliebrecht** [open-webui/open-webui#1063] [Q&A]
I am currently at 1.105 and in the about section it says an upgrade to 1.108 is available.   Ollama is currently 0.1,28

I installed without docker etc.

I am trying to upgrade to 1.108.  

I tried this and it successfully runs but I remain at 1.105:     (Sorry I am new to all of this, I am likely doing this incorrectly)

git clone https://github.com/open-webui/open-webui.git
cd open-webui/

# Copying required .env file
cp -RPp .env.example .env

# Building Frontend Using Node
npm i
npm run build

# Serving Frontend with the Backend
cd ./backend
pip install -r requirements.txt -U
bash start.sh

**Comment by justinh-rahb** [comment]
In the root of the open-webui directory you should run this before trying everything from `npm i` onward again:
```
git pull
```

**Reply by dliebrecht** [reply]
Here is the output:  ~/open-webui$ git pull
remote: Enumerating objects: 82, done.
remote: Counting objects: 100% (77/77), done.
remote: Compressing objects: 100% (33/33), done.
remote: Total 82 (delta 47), reused 60 (delta 40), pack-reused 5
Unpacking objects: 100% (82/82), 20.78 KiB | 1.89 MiB/s, done.
From https://github.com/open-webui/open-webui
   3455f89..20284dc  dev        -> origin/dev
Already up to date.

I ran all of the other subsequent commands with success.  I do get this error at the end:
No WEBUI_SECRET_KEY provided
Loading WEBUI_SECRET_KEY from .webui_secret_key
INFO:     Started server process [35618]
INFO:     Waiting for application startup.

#------------------------------------------------------------#
#                                                            #
#           'It would help me if you could add...'            #
#        https://github.com/BerriAI/litellm/issues/new        #
#                                                            #
#------------------------------------------------------------#

 Thank you for using LiteLLM! - Krrish & Ishaan



Give Feedback / Get Help: https://github.com/BerriAI/litellm/issues/new


INFO:     Application startup complete.
ERROR:    [Errno 98] error while attempting to bind on address ('0.0.0.0', 8080): address already in use
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.

**Reply by justinh-rahb** [reply]
> ```
> ERROR: [Errno 98] error while attempting to bind on address ('0.0.0.0', 8080): address already in use
> ```

The error is pretty clear: you're already running something on port 8080 (perhaps the previous attempt of running the WebUI).

**Reply by dliebrecht** [reply]
In an attempt to shutdown Open-webui I tried the following commands and then proceeded with the previous instructions starting with a GIT PULL.   I still am on Version 105 according to the about section.  (I am likely missing something simple.  (Very new) 

flipper@ML01:~/open-webui/backend$ sudo lsof -i :8080
[sudo] password for flipper:
COMMAND PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
uvicorn 566 root   16u  IPv4  39977      0t0  TCP *:http-alt (LISTEN)
flipper@ML01:~/open-webui/backend$ sudo kill 566
flipper@ML01:~/open-webui/backend$ sudo lsof -i :8080
COMMAND   PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
uvicorn 52069 root   16u  IPv4  19830      0t0  TCP *:http-alt (LISTEN)
flipper@ML01:~/open-webui/backend$ sudo kill 52069
flipper@ML01:~/open-webui/backend$ sudo lsof -i :8080

**Reply by justinh-rahb** [reply]
@dliebrecht When you're doing the npm commands, you're in the root of the open-webui directory, and not `backend/`, correct?

**Comment by dliebrecht** [comment]
It appears so - Output below:

flipper@ML01:~/open-webui$ npm i
npm run build

up to date, audited 298 packages in 13s

65 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> open-webui@0.1.108 build
> vite build

**Reply by justinh-rahb** [reply]
@dliebrecht alright next to check is if your browser is caching the frontend too aggressively.

**Reply by dliebrecht** [reply]
Cleared entire cache in Microsoft Edge Browser.  (All check boxes)  Still on 105.  
After clearing cache it asked me to login again which indicates cache cleared successfully.

**Reply by dliebrecht** [reply]
Not sure if this matters.  I am running Windows 11 with Ubuntu resident within WSL2.

**Reply by justinh-rahb** [reply]
Then it seems there has to be some confusion about installation method or location. The only way I will be able to sort this out is if you can give me a full accounting of exactly every step you've taken thus far, whether it worked or not, so I can work out where things went off the rails. Failing that, you could always start removing things and start from step zero.

**Reply by justinh-rahb** [reply]
Also this entire discussion is exactly why we encourage the use of Docker. It removes most of this ambiguity.

**Reply by dliebrecht** [reply]
Thank you for the help.  Will start again with Docker.  I really appreciate your time!

**Reply by a-atieh** [reply]
try this:

pip install --upgrade open-webui

hope it works!

**Reply by DemonRYB** [reply]
> try this:
> 
> pip install --upgrade open-webui
> 
> hope it works!

It actually works without Docker.
I only used Conda.
